HW1/P6/P6.py

```diff
@@ -0,0 +1,35 @@
+import findspark
+findspark.init()
+import pyspark
+sc = pyspark.SparkContext(appName="")
+import re
+import random
+
+## Note that to I have excluded the licensen information from the downloaded for text for better accuracy
+## Pre-process the text to create a list of words without all caps, all caps with an ending period, and all nums
+llist = sc.textFile('source.txt')
+wlist = llist.flatMap(lambda x: x.split())
+wlist_filtered = wlist.filter(lambda x: (not x[:-1].isupper()) if x[-1] == '.' else ((not x.isupper()) and (not x.isdigit())))
```

Comment:
I believe you can just check if x.isupper() or x.isdigit()

"HELLO.".isupper() returns True, so the additional check for the period seems repetitive